internal/encryption/keyprovider/errors.go

```diff
@@ -0,0 +1,66 @@
+package keyprovider
+
+import "fmt"
+
+// ErrKeyProviderFailure indicates a generic key provider failure.
+type ErrKeyProviderFailure struct {
+	Message string
+	Cause   error
+}
+
+func (e ErrKeyProviderFailure) Error() string {
+	if e.Cause != nil {
+		return fmt.Sprintf("%s (%v)", e.Message, e.Cause)
```

Comment:
If the underlying error will also have a message and cause, it will be endless nested parentheses.

Usually in Go you'd just use colon for the cause, so `%s: %v`. Also, I'm not sure if you want %v here or %s.

Reply:
But if you have a strong opinion about the parentheses, I'm fine with that.

Reply:
I was under the impression that the "endless parentheses" is a Go thing. Also no strong opinion.

Reply:
If this is not a strong "UX opinion" then I'd suggest going with `%s: %s` to avoid the endless parentheses. It's also what the errors.Wrap always did.